src/main/flight/gps_rescue.c

```diff
@@ -293,12 +301,26 @@ static void rescueAttainPosition()
 
 static void performSanityChecks()
 {
+    static uint32_t previousTimeUs = 0; // Last time Stalled/LowSat was checked
+    static int8_t secondsStalled = 0; // Stalled movement detection
+    static int8_t secondsLowSats = 0; // Minimum sat detection
+
     if (rescueState.phase == RESCUE_IDLE) {
         rescueState.failure = RESCUE_HEALTHY;
 
         return;
     }
 
+    if (rescueState.phase == RESCUE_INITIALIZE) {
```

Comment:
Combining this condition with the previous `rescueState.phase == RESCUE_IDLE` into a `switch` statement will look cleaner.

Reply:
The problem with the switch-case code in this particular situation is that return statements can be easily overlooked:

    switch (rescueState.phase) {
        case RESCUE_IDLE:
            rescueState.failure = RESCUE_HEALTHY;
            return;
     
        case RESCUE_INITIALIZE:
            [...]
            return;
    }

Reply:
Ok, I'll accept that. But at least the `if` should be combined as an `else if` with the previous condition.

Reply:
Ok, I've squashed and rebased with last upstream changes here: https://github.com/TonyBlit/betaflight/commit/a492b8d47d599401b2f0acbe6bca505a3c166591